hive_metastore_client/hive_metastore_client.py

```diff
@@ -131,7 +131,11 @@ def add_partitions_to_table(
             table_partition_keys=table.partitionKeys,
         )
 
-        self.add_partitions(partition_list_with_correct_location)
+        try:
+            self.add_partitions(partition_list_with_correct_location)
+            return True
+        except AlreadyExistsException:
```

Comment:
what if we added a log message here? for later debugging purposes

Reply:
Since this lib is open source, we preferred not to log in the lib and let this responsibility to the lib's users if necessary. This way we avoid that the lib does more than expected. For this reason, we added the returns indicating the operation results, so the user decides what to do with this operation status.
In our case, we added this log in our project that uses the lib.

Reply:
I see! Well, I'm not sure I agree 100% on letting the responsibility to the lib's users, cause in this case we need to expect that:
1) the callers would replicate a logger every time;
2) they need to enter the code just to make sure they understand what's a False return means.

of course number 2 can be solved by a better function description (either way, I think we should consider adding more details), but not sure the statement that the lib would do more than expected is concrete, because logging provides understanding and traceability despite what's being returned or coded.
we can see the logging as the comments you've added throughout  the code: it does more than just execute code, it works as a documentation for coding. So, logging is a documentation for execution and a way, besides messing with the context or logic (return boolean, for example), to tell what's going on.


what are your thoughts?

Reply:
Right, I think it does make sense. I have my concerns about security when logging some information in the lib that the user don't know is being logged as database or table names.
For this log, this wouldn't be a problem, but we should keep that in mind when adding logging in the lib.
Let's discuss it in slack.

Reply:
So, as we talked we'll analyze a logging framework to use in the lib and for now, we'll keep it without logs.